src/etc/inc/gwlb.inc

```diff
@@ -864,6 +869,15 @@ function return_gateways_array($disabled = false, $localhost = false, $inactive
 		$gateways_arr['Null4'] = $gwlo4;
 		$gateways_arr['Null6'] = $gwlo6;
 	}
+
+	if ($integer_index) {
+		$gateways_arr_temp = array();
```

Comment:
It is not so fussy as I thought - system_gateways and system_gateways_edit use the same array returned by return_gateways_array() to index the 'id' parameter to know which gateway to edit. As long as return_gateways_array() is consistent, all is good anyway.
system_gateways_edit finds the "real" gateway index to update in the config by:
```
$realid = $a_gateways[$id]['attribute'];
```
This looks not as critical as I had thought it might be.